### DISCUSSION TITLE
How scraping?

### DISCUSSION THREAD

**Discussion by shippokun** [markuplint/markuplint#446] [Q&A]
Hi!

I saw this tweet and asked a question.
How do you do your scraping?

https://twitter.com/markuplint/status/1516592390890397698?s=20&t=C7smbyhWSZEI4_oXAMrHUw
> they scrape MDN websites

I would appreciate it if you could let me know if you don't mind, as I may be able to help automate the process.
thank you.

---

こんにちは！

ツイートを拝見して質問をさせていただきました。
どのようにしてスクレイピングをしていますか？

https://twitter.com/cloud10designs/status/1516594133389185025?s=20&t=C7smbyhWSZEI4_oXAMrHUw
> MDNや仕様を直接スクレイピングして

自動化のお手伝いができるかもしれないので、差し支えがなければ教えていただけると幸いです。

**Comment by YusukeHirao** [comment]
@shippokun 

ありがとうございます。

スクレイピングの処理はNode.jsベースのcheerioを使って愚直にデータを取っています。

https://github.com/markuplint/markuplint/blob/84a541ebeab0404d1a440e81853f722ecade4c25/packages/%40markuplint/html-spec/gen/index.ts

で行なっていて、

https://github.com/markuplint/markuplint/blob/84a541ebeab0404d1a440e81853f722ecade4c25/package.json#L17

https://github.com/markuplint/markuplint/blob/84a541ebeab0404d1a440e81853f722ecade4c25/packages/%40markuplint/html-spec/package.json#L15

で NPM Script を実行するようにしています。

最終的にこのJSONファイルを生成するスクリプトになっていて、

https://github.com/markuplint/markuplint/blob/84a541ebeab0404d1a440e81853f722ecade4c25/packages/%40markuplint/html-spec/index.json

実行後にGit上の差分を確認し、その内容を見ながら以下のJSONファイルを手動で更新するようにしています。

https://github.com/markuplint/markuplint/tree/main/packages/%40markuplint/html-spec/src

**Reply by shippokun** [reply]
@YusukeHirao 
回答ありがとうございます。

追加で質問ですが、JSONの差分をチェックするタイミングは任意なのでしょうか？
→ GH Actions はスケジュールで定期実行することができるので、たとえば週1でスクレイピングを実行し、差分があればPRを出すことも可能です
更新するかどうかの判断はわからないのでPRで差分があるのかチェックして、追加で修正が必要ならIssueを立てるなりで後続に繋げるのが無難かと思っています

**Reply by YusukeHirao** [reply]
@shippokun 

差分チェックのタイミングは、現在は任意です。なるべく週1以内には…と完全に私の都合だけで動いています。
なので、Actionsが定期実行してくれるなら大変助かります。
更新するかどうかや修正箇所の判断は結局人の判断なので、おっしゃるとおりIssueを立てるだけで全然OKです！
